test/fit_tests/common/fit_common.py

```diff
@@ -703,18 +707,27 @@ def apply_obm_settings_new():
                             nodestatus[node]['status'] = "succeeded"
                         if state in ["failed", "cancelled", "timeout"]:
                             nodestatus[node]['status'] = "pending"
-        if VERBOSITY > 4:
+                            # if the workflow left an invalid OBM, delete it
+                            result = rackhdapi("/api/2.0/nodes/" + node)
+                            if result['status'] == 200:
+                                if result['json']['obms']:
+                                    for ref in result['json']['obms']:
+                                        obmref = ref.get('ref')
+                                        if obmref:
+                                            rackhdapi(obmref, action="delete")
+
+        if VERBOSITY >= 4:
             print "**** Node(s) OBM status:\n", json.dumps(nodestatus, sort_keys=True, indent=4,)
         if "pending" not in str(nodestatus) and "running" not in str(nodestatus):
             # All OBM settings successful
             return True
-        time.sleep(10)
+        time.sleep(30)
```

Comment:
Am I reading the code correctly that using this functional will mandate a minimum time to invoke this method as being `30` seconds (from this `time.sleep` function) times the 30 count above, making it take 15 minutes to ever invoke this method, regardless of how long it takes the various workflows to complete?

Can we improve this to optimistically escape the range count if all the workflows come back with a "workflow complete, success" marker?

Is there any intent that a failure of one of the workflows is exposed from within this function beyond the logging? (i.e. is the intent here that this is a setup/utility method that will work 99% of the time, or is this the core of a use case/test that would be better served with more explicit error information when it fails?)